### DISCUSSION TITLE
Replacement for the get_response_synthesizer(response_mode="no_text") python method

### DISCUSSION THREAD

**Discussion by matzutt** [run-llama/LlamaIndexTS#1250] [Ideas]
In our company's project we previously used the python version of LlamaIndex but are currently migrating to TS. 
We used the `get_response_synthesizer(response_mode="no_text")` method, because we just needed to get the nodes in a similarity search back, which essentially returns a responseSynthesizer that just returns the nodes.

In order to have a similar effect in TS we extended the BaseSynthesizer class as follows
```js
import { ModuleRecord, PromptsRecord, SynthesizeQuery, LLM, MessageContent, PromptHelper, BaseSynthesizer } from 'llamaindex';
import {EngineResponse, NodeWithScore} from '@llamaindex/core/schema';

class NoTextSynthesizer extends BaseSynthesizer {
  declare llm: LLM;
  declare promptHelper: PromptHelper;

  constructor() {
    super({llm: 'dummy' as unknown as LLM, promptHelper: new PromptHelper()});
  }

  getResponse(query: MessageContent, textChunks: NodeWithScore[], stream: boolean): Promise<EngineResponse | AsyncIterable<EngineResponse>> {
    throw new Error('Method not implemented.');
  }
  validatePrompts(promptsDict: PromptsRecord, moduleDict: ModuleRecord): void {
    throw new Error('Method not implemented.');
  }
  getPrompts(): PromptsRecord {
    throw new Error('Method not implemented.');
  }
  updatePrompts(prompts: PromptsRecord): void {
    throw new Error('Method not implemented.');
  }
  protected _getPrompts(): PromptsRecord {
    throw new Error('Method not implemented.');
  }
  protected _updatePrompts(prompts: PromptsRecord): void {
    throw new Error('Method not implemented.');
  }
  protected _getPromptModules(): ModuleRecord {
    throw new Error('Method not implemented.');
  }

  synthesize(query: SynthesizeQuery, stream: true): Promise<AsyncIterable<EngineResponse>>;
  synthesize(query: SynthesizeQuery, stream: false): Promise<EngineResponse>;
  async synthesize(query: SynthesizeQuery, stream: boolean): Promise<AsyncIterable<EngineResponse> | EngineResponse> {
    const { nodes } = query;

    const response = '';
    return EngineResponse.fromResponse(response, false, nodes);
  }
}

export { NoTextSynthesizer };
```

If this is interesting for more people but us, we are happy to contribute the code.
(and yes, I know that the dummy llm casting is not ideal, but this is just a last minute adaption to changes in the current llamaIndexTS release).

If there is a better way to get a similar effect, we would be also interested.

**Comment by himself65** [comment]
We don't have it right now, feel free to open a PR for it!

**Reply by himself65** [reply]
https://github.com/run-llama/LlamaIndexTS/blob/main/packages/core/src/response-synthesizers/factory.ts

**Reply by matzutt** [reply]
Thanks, happy to beeing able to contribute. I will try to adopt our changes to suit the factory method you referred to, soonishly. 
Because I have not yet worked with 'changesets'. Is the way to go to generate an issue out of this discussion, create a branch for the issue and then pose a PR (supposed all is building and working well)?
